Dockerfile

```diff
@@ -1,22 +1,70 @@
-# syntax=docker/dockerfile:1
-# https://github.com/docker/docs/issues/20935
+# syntax=docker.io/docker/dockerfile:1
 
-FROM node:22-alpine
+# --------------------------------------------
+# Base stage
+# --------------------------------------------
+FROM node:22-alpine AS base
 
+# --------------------------------------------
+# Dependencies stage
+# --------------------------------------------
+FROM base AS deps
+RUN apk add --no-cache libc6-compat
 WORKDIR /app
 
-COPY package.json package-lock.json ./
-RUN npm install
+# Copy package management files
+COPY package.json pnpm-lock.yaml .npmrc* ./
 
+# Since we are using corepack, we need to modify the pnpm cache dir's path
+RUN corepack enable pnpm && \
+    pnpm config set store-dir /pnpm/store
+
+# Install dependencies with cache optimization
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store \
+    pnpm i --frozen-lockfile
+
+# --------------------------------------------
+# Builder stage
+# --------------------------------------------
+FROM base AS builder
+WORKDIR /app
+
+# Copy dependencies and source code
+COPY --from=deps /app/node_modules ./node_modules
 COPY . .
 
-# https://docs.docker.com/build/building/secrets/
-RUN --mount=type=secret,id=sentry_auth_token,env=SENTRY_AUTH_TOKEN \
```

Comment:
Sentry auth token is required during build time so that the sentry plugin can generate and upload source map to Sentry